can/parser_pyx.pyx

```diff
@@ -49,8 +49,7 @@ cdef class CANParser:
     self.can_invalid_cnt = CAN_INVALID_CNT
 
     cdef int i
-    cdef int num_msgs = self.dbc[0].num_msgs
-    for i in range(num_msgs):
+    for i in range(self.dbc[0].msgs.size()):
```

Comment:
usually the cdef is to be explicit about being a C loop, did you check if this is this still a C loop?

Reply:
main difference I see is `__Pyx_decode_cpp_string` with this PR vs `__Pyx_decode_c_string` with master for the decode line

Reply:
Definitely gets slower without the `cdef i`. avg `0.10887859389185905` ms vs `0.16389694064855576` ms

Reply:
Ah there's no more strlen called on the string when decoding, no matter if cdef is there or not. Think it's not a c loop anymore